postreise/plot/plot_states.py

```diff
@@ -65,109 +65,156 @@ def get_state_borders():
-    leg_color,
-    leg_label,
-    font_size,
+    state_list=None,
+    colors=None,
+    line_color="grey",
+    line_width=1,
+    fill_alpha=0.8,
+    labels_list=None,
+    legend_colors=None,
+    legend_labels=None,
+    legend_title=None,
+    font_size="10pt",
+    bokeh_figure=None,
     web=True,
     us_states_dat=None,
+    background_map=True,
 ):
     """Plots US state borders and allows color coding by state,
         for example to represent different emissions goals.
 
     :param list state_list: list of us states to color code.
-    :param list color_list: list of colors associated with states in state_list.
-    :param list labels_list: list of labels for us states.
-    :param list leg_color: list of colors for legend
-    :param list leg_label: list of labels for colors in legend
+    :param str/iterable colors: color or colors associated with states in state_list.
+    :param str line_color: color of state outlines.
+    :param int/float line_width: thickness of state outlines.
+    :param int/float fill_alpha: fill alpha for state patches.
+    :param iterable labels_list: list of labels for us states.
+    :param iterable legend_colors: list of colors for legend.
+    :param iterable legend_labels: list of labels for colors in legend.
+    :param str legend_title: title for legend.
     :param float font_size: citation font size, for non web version only
+    :param bokeh.plotting.figure.Figure bokeh_figure: figure to plot onto, if provided.
+        If not, a new figure will be created.
     :param bool web: if true, formats for website with hover tips
     :param dict us_states_dat: dictionary of state border lats/lons. If None, get
         from :func:`postreise.plot.plot_states.get_state_borders`.
+    :param bool background_map: if True, plot a map background behind state borders.
+    :raises TypeError: if inputs are wrong type.
+    :raises ValueError: if inputs have wrong values.
     :return: (*bokeh.plotting.figure.Figure*) -- map of us states with option to color
         by value.
     """
-    # warn if inputs diff lengths
-    if len(state_list) != len(color_list):
-        print("warning: state_list and color_list must be same length")
-    if len(state_list) != len(labels_list):
-        print("warning: state_list and labels_list must be same length")
-    # Get state borders if necessary, and project to "EPSG:3857" either way
+    # Get state borders data if necessary
     if us_states_dat is None:
         us_states_dat = get_state_borders()
+    if state_list is None:
+        # Default to continental US from state borders data, w/ no colors, no labels.
+        if labels_list is not None:
+            raise TypeError("Cannot specify labels_list without state_list")
+        if colors is not None and not isinstance(colors, str):
+            raise TypeError("Cannot specify list of colors without state_list")
+        state_list = list(set(us_states_dat.keys()) - {"AK", "HI", "DC", "PR"})
```

Comment:
If we just do `state_list = list(set(us_states_dat.keys()) - {"AK", "HI", "DC", "PR"}) if state_list is None`, then we don't have to do the checks on `labels_list` and `colors`, no?

Reply:
I don't think so, because dict keys are unordered but the `labels_list` and `colors` _are_ ordered, so the user can't pass these and have them map properly if they don't also pass the `state_list`.

Reply:
I see

Reply:
I guess you thought about of grouping `state_list`, `colors` and `labels_list` in a dict that the user would pass as a parameter, e.g. `{"California": ["blue", "CA"]}`

Reply:
That would work too. There is no obligation, I was just trying to simplify the checks but it is transparent to the user so whatever is pick whatever is best from a user experience.

Reply:
> I hadn't, but it's not a bad idea. Or we could take in a pandas DataFrame that would have either:
> 
>     * an index but no columns (the index says which states we should plot)
> 
>     * states and index plus colors and/or labels as columns
> 
> 
> Then we wouldn't have to do length checks.
> 
> Mostly, I wanted to be able to plot with none of these specified, to serve other plots that need borders but nothing else. See the current `daniel/powerflow_snapshot` for an example which uses: `plot_states(bokeh_figure=bokeh_figure, background_map=background_map)` and then builds further things on top.

so leave it as is